akka-actor-typed/src/main/scala/akka/actor/typed/BehaviorInterceptor.scala

```diff
@@ -17,6 +17,13 @@ import akka.annotation.{ DoNotInherit, InternalApi }
 abstract class BehaviorInterceptor[O, I] {
   import BehaviorInterceptor._
 
+  /**
+   * Allows for bypassing the interceptor for certain messages. Useful if the official protocol and the actual
+   * protocol of an actor causes problems, for example class cast exceptions for a message not of type `O` that
+   * the actor still knows how to deal with.
+   */
+  def applyFor(message: Any): Boolean = true
```

Comment:
Trying to grep why the message should be allowed through of not `O`, doesn't it leak the idea of how the typed actor idea is constructed or are we saying we now want something else.

Reply:
Or perhaps we want a strongly typed and a declarable but separate middle ground actor that would allow and include this - vs having it available to be used here.

Reply:
kind of `isDefinedAt`,isn't it?

Reply:
I don't know exactly what the reason was but I recall that it was something that was suggested in the review but said that it didn't work.

Reply:
A better API might be:
```
def interceptMessageType[M]: Class[M <: O]
```

then it's more like an optional ClassTag

Reply:
More for the specific case but less powerful, because the current one allows for arbitrary selection of messages to select. Is it really better with the class one?

Reply:
arbitrary selection can be done inside the `override def aroundReceive`

Reply:
Fair enough